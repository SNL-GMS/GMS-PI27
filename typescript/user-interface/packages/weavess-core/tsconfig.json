{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./lib",
    "strictNullChecks": true
  },
  "include": ["src/**/*", "__tests__/**/*", "__mocks__/**/*"]
}
