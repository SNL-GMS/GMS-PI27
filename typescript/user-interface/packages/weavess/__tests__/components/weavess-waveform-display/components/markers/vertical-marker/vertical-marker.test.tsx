/* eslint-disable react/jsx-props-no-spreading */
import { WeavessTypes } from '@gms/weavess-core';
import { render } from '@testing-library/react';
import * as React from 'react';

import type { VerticalMarkerProps } from '../../../../../../src/ts/components/weavess-waveform-display/components/markers/vertical-marker/types';
import { VerticalMarker } from '../../../../../../src/ts/components/weavess-waveform-display/components/markers/vertical-marker/vertical-marker';

const props: VerticalMarkerProps = {
  name: 'my-marker',
  color: '#ff0000',
  lineStyle: WeavessTypes.LineStyle.DASHED,
  percentageLocation: 90
};
describe('Weavess Vertical Marker', () => {
  it('Weavess Vertical Marker to be defined', () => {
    expect(VerticalMarker).toBeDefined();
  });

  it('renders', () => {
    const { container } = render(<VerticalMarker {...props} />);
    expect(container).toMatchSnapshot();
  });
});
