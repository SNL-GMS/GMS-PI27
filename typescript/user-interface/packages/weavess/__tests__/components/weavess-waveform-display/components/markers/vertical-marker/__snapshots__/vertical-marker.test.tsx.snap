// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weavess Vertical Marker renders 1`] = `
<div>
  <div
    class="vertical-marker"
    data-cy="weavess-static-vertical-marker"
    data-cy-color="#ff0000"
    data-cy-left="90"
    style="left: calc(90% - 0px); width: 1px; border-inline-end: #ff0000 dashed 1px; transform: translateX(-50%);"
  />
</div>
`;
